provide.tests/NChainTest.cs

```diff
@@ -14,27 +18,107 @@ private async Task<string> CreateIdentForTestUser()
+            {
                 FirstName = "Test",
                 LastName = "User",
                 Email = email,
                 Password = "prvdp4ss",
             });
 
             return (await Ident.Authenticate(
-                new Auth 
+                new Auth
                 {
                     Email = email,
                     Password = "prvdp4ss",
                 }
             )).Token.Token;
         }
 
-        // [Fact]
-        // public async void TestInitNChain() 
-        // {
-        //     var token = await CreateIdentForTestUser();
-        //     var nchain = NChain.InitNChain(token);
-        // }
+        [Fact]
+        public async Task TestInitNChain()
+        {
+            // same way of getting compiled artifacts as in bri-1 example
+            var httpClient = new HttpClient();
+            var request = new HttpRequestMessage(new HttpMethod("get"), "https://s3.amazonaws.com/static.provide.services/capabilities/provide-capabilities-manifest.json");
+            var res = await httpClient.SendAsync(request);
+            var content = res.Content;
+
+            var raw = await content.ReadAsByteArrayAsync();
+            var str = System.Text.Encoding.Default.GetString(raw);
+            var parsed = JObject.Parse(str);
+            var contract = parsed["baseline"]["contracts"][2];
+
+            var token = await CreateIdentForTestUser();
+            var ident = Ident.InitIdent(token);
+
+            var organization = await ident.CreateOrganization(new Organization
+            {
+                Name = "test organization"
+            });
+
+            var application = await ident.CreateApplication(new Application
+            {
+                Name = "test application",
+            });
+
+            var accessRefreshToken = await ident.CreateToken(new JWTToken
+            {
+                Scope = "offline_access",
+                ApplicationId = application.Id
+            });
+
+            ident = new Ident(accessRefreshToken.AccessToken);
+            
+            // FEEDBACK NEEDED: Do we need something else to add orgs to org registry?
+            // how do we get org address to id org in contract?
+            await ident.AddApplicationOrganizations(application.Id.ToString(), new ApplicationOrganization
+            {
+                OrganizationId = organization.Id.Value
+            });
+
+            // this network needs to be activated in db, at least that is done first in bri-1 example
+            var ethTestNetId = "66d44f30-9092-4182-a3c4-bc02736d6ae5";
+            var nchain = NChain.InitNChain(accessRefreshToken.AccessToken);
+
+            await nchain.CreateAccount(new Model.NChain.Account
+            {
+                NetworkId = new Guid(ethTestNetId),
+            });
+            var accounts = await nchain.ListAccounts(new Dictionary<string, object>());
+
+            await nchain.CreateContract(new Model.NChain.Contract

```

Comment:
Can we just assume that the OrgRegistry contract is already deployed somewhere and just query it for organization nats address?

Reply:
Contract creation for every test run seems like too much time and effort